sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1700,6 +1700,16 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val VIEW_SCHEMA_BINDING_MODE = buildConf("spark.sql.viewSchemaBindingMode")
+    .doc("Set to DISABLE to disable the WITH SCHEMA clause for view DDL and suppress the line in " +
+      " DESCRIBE EXTENDED. The default, and only other value, is COMPENSATION. Views without " +
+      " WITH SCHEMA clause are defaulted to WITH SCHEMA COMPENSATION.")
+    .version("4.0.0")
+    .stringConf
+    .transform(_.toUpperCase(Locale.ROOT))
+    .checkValues(Set("COMPENSATION", "DISABLED"))
```

Comment:
why don't we allow other modes to be the default?

Reply:
We did, @gengliangwang raised the question whether that is needlessly complex.
I have been splitting the baby by removing all but these two. having on/off config which is not a boolean. 
If users demand it we can easily expand the config to more values, if not we can deprecate the the config and remove it.

Reply:
Shall we also support only `COMPENSATION` in the `WITH SCHEMA` clause? We should make them consistent.

Reply:
If you prefer having 5 modes in the WITH SCHEMA clause, let's have 5 modes in the SQL Conf as well. Sorry for causing the misunderstanding.